Directory.Build.props

```diff
@@ -210,6 +210,12 @@
   </PropertyGroup>
 
   <PropertyGroup>
+    <!-- Disable crossgen on:
+          * NetBSD, illumos and Solaris for now. This can be revisited when we have full support.
+          * FreeBSD when cross building from Linux. -->
+    <DisableCrossgen2 Condition="'$(TargetOS)'=='NetBSD' or '$(TargetOS)'=='illumos' or '$(TargetOS)'=='Solaris' or
+        ('$(TargetOS)'=='FreeBSD' and '$(CrossBuild)'=='true')">true</DisableCrossgen2>
```

Comment:
Opened #66866 to use live-build of apphost on these platforms.

cc @ViktorHofer, @agocke, ideally we should _always_ use the live apphost for shipping tools, unless there is an issue with that?